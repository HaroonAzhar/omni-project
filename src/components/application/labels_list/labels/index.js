export { default as WarningLabel } from "./warning_label";
