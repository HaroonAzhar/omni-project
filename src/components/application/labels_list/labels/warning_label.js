/* eslint-disable jsx-a11y/label-has-for */
import React from "react";
import PropTypes from "prop-types";

import { Label } from "components/atoms";

const labelText =
  "Step(s) need rechecking as the DIP was edited since these steps were last amended";

const WarningLabel = ({ applicationSteps = [] }) => {
  const shouldShowLabel = applicationSteps.some(
    ({ status }) => status === "Recheck"
  );

  return shouldShowLabel ? <Label text={labelText} color="warn" /> : null;
};

WarningLabel.propTypes = {
  applicationSteps: PropTypes.array,
};

export default WarningLabel;
