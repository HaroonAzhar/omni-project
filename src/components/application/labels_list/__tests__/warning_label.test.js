import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";

import { WarningLabel } from "../labels";

const getComponent = (applicationSteps) => {
  return (
    <ThemeProvider theme={{ colors: { warn: "red", darkWarn: "blue" } }}>
      <WarningLabel applicationSteps={applicationSteps} />
    </ThemeProvider>
  );
};

describe("<WarningLabel>", () => {
  it("Renders correctly", () => {
    const application = [{ status: "Recheck" }];

    const tree = renderer.create(getComponent(application)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Renders when status "recheck" occured', () => {
    const application = [
      { status: "New" },
      { status: "Recheck" },
      { status: "New" },
    ];

    const { getByText } = render(getComponent(application));
    const labelText =
      "Step(s) need rechecking as the DIP was edited since these steps were last amended";

    expect(getByText(labelText)).toBeDefined();
  });

  it("Doesn't render when there isn't \"recheck\" status", () => {
    const application = [{ status: "New" }];

    const tree = renderer.create(getComponent(application)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
