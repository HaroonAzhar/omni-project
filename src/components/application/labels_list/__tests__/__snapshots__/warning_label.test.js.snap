// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarningLabel> Doesn't render when there isn't "recheck" status 1`] = `null`;

exports[`<WarningLabel> Renders correctly 1`] = `
<span
  className="sc-gzVnrw dvDegH"
  color="warn"
>
  Step(s) need rechecking as the DIP was edited since these steps were last amended
</span>
`;
